src/l500/l500-serializable.cpp

```diff
@@ -18,93 +18,112 @@ namespace librealsense
 
     std::vector<uint8_t> l500_serializable::serialize_json() const
     {
-        json j;
-        auto options = _depth_sensor.get_supported_options();
-
-        for (auto o : options)
-        {
-            auto&& opt = _depth_sensor.get_option(o);
-            auto val = opt.query();
-            j[get_string(o)] = val;
-        }
-
-        auto str = j.dump(4);
-        return std::vector<uint8_t>(str.begin(), str.end());
+        auto l500_uvc_sensor = std::dynamic_pointer_cast<uvc_sensor>(_depth_sensor.get_raw_sensor());
+        if (!l500_uvc_sensor)
+            throw invalid_value_exception(
+                to_string() << "Failed to get raw depth sensor from serializable device" );
```

Comment:
I think the exception isn't needed -- just don't call invoke_powered?

Reply:
Up to you

Reply:
That also mean that we don't do the action, invoke power get the inner code and run it

Reply:
You can call the lambda manually.
But up to you.

Reply:
This code was removed